### PR TITLE
Fixing height and width of button when showLoader is true

### PR DESCRIPTION
**Pull request by joeldenning**
See https://canopytax.atlassian.net/browse/CON-298

### CODE REVIEW

File: src/custom-elements/cps-button/cps-button.js [hunk 1]
```diff
@@ -64,11 +63,21 @@ class CpsButton extends Component {
 		}
 	}
 	prepareElementForLoader = () => {
-		this.widthBeforeLoader = this.props.customElement.style.clientWidth;
```
Comment (joeldenning) [inline src/custom-elements/cps-button/cps-button.js]: `element.style.clientWidth` is not a thing. That was the bug. Instead it should be `element.clientWidth`

File: src/custom-elements/cps-button/cps-button.js [hunk 2]
```diff
@@ -31,9 +31,8 @@ class CpsButton extends Component {
 
 		// Only return something if we want to completely overwrite the innerHTML
 		if (this.props.showLoader) {
-			if (!this.props.customElement.querySelector('.cps-loader')) {
-				this.prepareElementForLoader();
-			}
+			this.prepareElementForLoader();
```
Comment (joeldenning) [inline src/custom-elements/cps-button/cps-button.js]: The whole purpose of this is to make sure that when someone sets the `showLoader` property to be true that the size of the button doesn't change. What we do is look at the button's previous width and height and make sure that when the loader appears that that height doesn't change.

File: src/custom-elements/cps-button/cps-button.js [hunk 3]
```diff
@@ -77,10 +86,14 @@ class CpsButton extends Component {
 		if (this.widthBeforeLoader) {
 			this.props.customElement.style.width = this.widthBeforeLoader;
 			this.props.customElement.style.height = this.heightBeforeLoader;
+		} else {
+			this.props.customElement.style.removeProperty('width');
```
Comment (joeldenning) [inline src/custom-elements/cps-button/cps-button.js]: We set the width and the height inside of `prepareElementForLoader`, now we need to remove them when showLoader is set to false.

File: src/custom-elements/cps-button/cps-button.js [hunk 4]
```diff
@@ -64,11 +63,21 @@ class CpsButton extends Component {
 		}
 	}
 	prepareElementForLoader = () => {
-		this.widthBeforeLoader = this.props.customElement.style.clientWidth;
-		this.heightBeforeLoader = this.props.customElement.style.clientHeight;
+		if (this.elementPreparedForLoader) {
```
Comment (joeldenning) [inline src/custom-elements/cps-button/cps-button.js]: Before we were doing a querySelector in the render. Now we just keep track of whether this needs to happen with a variable on the class. This avoids touching the dom every render.

**Comment by TheMcMurder** [comment]
🐙

**Pull request merged**